crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -119,10 +124,149 @@ impl Default for ECMHLiveObjectSetDigest {
     }
 }
 
+// TODO: Does this need versioning?
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ObjectCheckpointState {
```

Comment:
I wonder if we should just use ObjectRef. It will remove a lot of this code, and also make it possible to prove that version N was written in a checkpoint. Technically this is already possible via the digest, but the proof would have to include the entire object contents (which includes the version).

Reply:
I think that makes a lot of sense. It should likely get rid of the extra Ord/PartialEq impl's as well since ObjectRef already implements them.